lib/OpenQA/Schema/Result/Jobs.pm

```diff
@@ -592,6 +593,9 @@ sub failed_modules_with_needles {
             for my $needle (@{$detail->{needles}}) {
                 push( @needles, [ $needle->{name}, $counter ] );
             }
+            if (!@{$detail->{needles}}) {
+                push(@needles, [ undef, $counter ] );
```

Comment:
Right, I was referring to the :594, where the space is. And also there is a space before last closing ). I personally don't mind which way we go. You have LGTM and later perltidy rule can be added to consolidate all. I'll create new issue in progress.